### PR TITLE
AC-2497::Credit card entry format is not descriptive. (Checkout - Pay…

### PR DESCRIPTION
**Pull request by glo71317**
## Description
Credit card entry format is not descriptive. (Checkout - Payment)

**Reproduction Steps**
select "Blouses & Shirts" > select "Susanna Draped Tank" > add product to cart > select bag > select "Edit Shopping Bag" > select "Proceed to Checkout" > complete shipping information and continue to "Shipping Method" > complete shipping method and continue to "Payment Information" (TEST - Payment information][NODE][body>div:nth-of-type(1)>*:nth-child(1)>*:nth-child(3)>*:nth-child(1)>*:nth-child(1)>*:nth-child(5)>*:nth-child(1)>*:nth-child(1)>*:nth-child(1)>*:nth-child(1)]

1. With JAWS operating, press Tab to move focus to the Card Number input.

**Actual Behavior**
The Card Number input has a placeholder value of 16 bullet characters. For users whose screen readers announce the placeholder attribute, this text is verbose and confusing.

**Expected Behavior**
Ensure that label and description text is appropriately descriptive. If a format hint is necessary, provide it in text that is persistently visible and programmatically associated with the input. Text such as "16-digit number" would be more appropriate than bullet characters.

## Related Issue
Closes https://jira.corp.magento.com/browse/AC-2497.

## Verification Steps
**Pre-Conditions:**

1. Have Magento instance with sample data installed
2. Make sure to have pwa studio installed
3. Make sure to have a customer login for front end login

**Manual Steps executed:**

Login to venia > Navigate to global header accessories or other
Add product to cart > select bag > select "Edit Shopping Bag" > select "Proceed to Checkout" > complete shipping information and continue to "Shipping Method" > complete shipping method and continue to "Payment Information" (TEST - Payment information] 
Enable your screen reader and press Tab to move focus to the Card Number input.

**✖️ Behaviour Before The Fix :** The Card Number input has a placeholder value of 16 bullet characters. For users whose screen readers announce the placeholder attribute, this text is verbose and confusing.
image
**✔️Behaviour After The Fix:** The Card Number input has a Text such as "16-digit number"
image

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### Resolved issues:
1. [x] resolves magento/pwa-studio#3833: AC-2497::Credit card entry format is not descriptive. (Checkout - Pay…

### CODE REVIEW

File: packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/brainTreeDropIn.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ const BraintreeDropin = props => {
                 overrides: {
                     fields: {
                         number: {
+                            placeholder: '16-Digit Number',
```
Comment (chris-brabender) [inline packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/brainTreeDropIn.js:55]: For internationalisation all strings should be translatable.

Comment (glo71317) [reply packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/brainTreeDropIn.js]: done, @chris-brabender have a look pls

**Comment by ericeoeur** [comment]
run all tests

**Comment by sharkySharks** [comment]
run pr-deploy

**Comment by haal9000** [comment]
run pr-deploy

**Comment by haal9000** [comment]
run pr-deploy

**Review by ericeoeur — APPROVED** [review]
@glo71317 The deployment issues for this ticket have been fixed and this has passed code review. However, see necessary edits below as edits did not pass ESlint. 

- [ ] `../CheckoutPage/PaymentInformation/brainTreeDropIn.js` ESLint. Did not pass ESLint. Please go ahead and execute yarn run lint locally for more details

**Comment by dpatil-magento** [comment]
run cypress

**Comment by dpatil-magento** [comment]
QA Approved, cypress failure not related.

**Comment by dpatil-magento** [comment]
@glo71317  Translation still not picking for french store.  Also when I try to fix lint it also throwing related error - 

```
pwa-studio/packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/brainTreeDropIn.js
  71:8  error  React Hook useCallback has a missing dependency: 'formatMessage'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

✖ 1 problem (1 error, 0 warnings) 

```

French store -

image

**Comment by abhattGlo** [comment]
@dpatil-magento the translation is not working without the fix as well.
as payment related fields are coming from braintree. We are looking if it can be fixed

**Comment by glo71317** [comment]
> QA Approved, cypress failure not related.

@dpatil-magento We checked and it is not working earlier also, scope of this ticket need to add the placeholder for card number, And also we had observed it is coming from Braintree directly hence translation is not working. 
So, Please suggest if it is really make it work then you can create separate ticket for the same and also not sure how it will be feasible to do that.

**Comment by dpatil-magento** [comment]
Translation working fine on local stage:venia so just might be aws env issue.

**Pull request merged**